homeassistant/components/wallbox/sensor.py

```diff
@@ -0,0 +1,99 @@
+"""Home Assistant component for accessing the Wallbox Portal API. The sensor component creates multiple sensors regarding wallbox performance."""
+
+from datetime import timedelta
+import logging
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import CONF_CONNECTIONS, DOMAIN, SENSOR_TYPES
+
+CONF_STATION = "station"
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def wallbox_updater(wallbox, station):
+    """Get new sensor data for Wallbox component."""
+    data = wallbox.getChargerStatus(station)
+    filtered_data = {k: data[k] for k in SENSOR_TYPES if k in data}
+
+    for key, value in filtered_data.items():
+        sensor_round = SENSOR_TYPES[key]["ATTR_ROUND"]
+        if sensor_round:
+            try:
+                filtered_data[key] = round(value, sensor_round)
+            except TypeError:
+                _LOGGER.debug("Cannot format %s", key)
+
+    return filtered_data
+
+
+async def async_setup_entry(hass, config, async_add_entities):
+    """Create wallbox sensor entities in HASS."""
+
+    wallbox = hass.data[DOMAIN][CONF_CONNECTIONS][config.entry_id]
+    station = config.data[CONF_STATION]
+
+    async def async_update_data():
+
+        try:
+            return await hass.async_add_executor_job(wallbox_updater, wallbox, station)
```

Comment:
It would be nice it wallbox_updater was async, so that you had the async_add_executor_job centralised in one py file, which makes it easier to update in the future.

Reply:
yes, I can make that change here.

Reply:
👍